### PR TITLE
Fix normalisation

### PR DESCRIPTION
**Pull request by tomMcGrath**
# Description

Anthropic's SAE training code normalises activations before they enter the SAE so that their average L2 norm is sqrt(d_model). Although we have normalisation activation in the code, it is not currently applied during evals which both harms the cross-entropy loss and makes hyperparameter transfer across layers impossible as different layers have substantially different mean norms.

This PR changes estimation of the scale factor to occur once, with a larger sample size, before SAE training, and modifies the eval code to up- and downscale activations as part of the hook function as well as correctly estimate the L2 norm of activations that the SAE will actually see during training.

The code in this PR isn't the prettiest and I aim to create a tidier version next week but we need this functionality. I've marked areas for tidying with `TODO(tomMcGrath)` and will open an issue to track this. EDIT: this is now tracked in issue #151.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### You have tested formatting, typing and unit tests (acceptance tests not currently in use)

- [x] I have run `make check-ci` to check format and linting. (you can run `make format` to format code if needed.)

### Performance Check.

If you have implemented a training change, please indicate precisely how performance changes with respect to the following metrics:
- [ ] L0
- [x] CE Loss
- [ ] MSE Loss
- [ ] Feature Dashboard Interpretability

[This W&B dashboard ](https://wandb.ai/tmcgrath/gpt-2-sweep-15may24-try-normalisation/workspace.)shows the effect of the fix:
- `metrics/l2_norm_in` shows the L2 norm of the inputs fluctuating around sqrt(d_model)
- `metrics/CE_loss_score` shows the effects of correctly up- and downsampling, as well as the effects of no normalisation (the state pre-fix) and only upscaling

### CODE REVIEW

**Comment by jbloomAus** [comment]
Awesome work! Thanks so much!
